nano/node/transport/transport.cpp

```diff
@@ -188,3 +203,46 @@ bool nano::transport::reserved_address (nano::endpoint const & endpoint_a, bool
 	}
 	return result;
 }
+
+nano::bandwidth_limiter::bandwidth_limiter (const size_t limit_a) :
+last_poll (std::chrono::steady_clock::now ()),
+limit (limit_a),
+rate (0),
+trended_rate (0)
+{
+}
+
+bool nano::bandwidth_limiter::should_drop (const size_t & message_size)
+{
+	using namespace std::chrono_literals;
+	bool result (false);
+	if (limit == 0) //never drop if limit is 0
+	{
+		return result;
+	}
+	std::lock_guard<std::mutex> lock (mutex);
+
+	if (last_poll + 50ms < std::chrono::steady_clock::now ())
```

Comment:
This seems like it would miss the case where a bunch of traffic has come in within the last 50ms interval.

Can this be implemented simply by dividing the limit/sec by (50ms / 1000ms) and then seeing if rate is greater?  If it is, drop, if it isn't, add message_size to rate and let it pass.